core/src/main/java/org/opensearch/sql/planner/optimizer/LogicalPlanOptimizer.java

```diff
@@ -52,7 +51,7 @@ public static LogicalPlanOptimizer create() {
          * Phase 2: Transformations that rely on data source push down capability
          */
         new CreateTableScanBuilder(),
-        new CreatePagingTableScanBuilder(),
+        new PushDownPageSize(),
```

Comment:
curious why we aren't using `TableScanPushDown.PUSH_DOWN_PAGESIZE`?  Is pagesize not a concept that the TableScan should know about?

Reply:
`PushDownPageSize` does not really fit into the simple pattern of "when node of this type is encountered, call this method" that other `PUSH_DOWN_*` rules use.

It's more similar to `CreateTableScanBuilder` rule.

Reply:
Does it similar to PUSH_DOWN_LIMIT? and should be at the end of rule list?

Reply:
Why should limit be at the end of the list? I did move `PushDownPageSize` to be after limit.